Analysis/Tasks/emcalCorrectionTask.cxx

```diff
@@ -0,0 +1,199 @@
+
+      mEmcalCells.emplace_back(o2::emcal::Cell(
+        cell.cellNumber(),
+        cell.amplitude(),
+        cell.time(),
+        o2::emcal::intToChannelType(cell.cellType())));
+    }
+
+    // Cell QA
+    // For convenience, use the clusterizer stored geometry to get the eta-phi
+    for (auto& cell : mEmcalCells) {
+      hCellE->Fill(cell.getEnergy());
+      hCellTowerID->Fill(cell.getTower());
+      auto res = mClusterizer->getGeometry()->EtaPhiFromIndex(cell.getTower());
+      hCellEtaPhi->Fill(std::get<0>(res), std::get<1>(res));
+      res = mClusterizer->getGeometry()->GlobalRowColFromIndex(cell.getTower());
+      // NOTE: Reversed column and row because it's more natural for presentatin.
+      hCellRowCol->Fill(std::get<1>(res), std::get<0>(res));
+    }
+
+    // TODO: Helpful for now, but should be removed.
+    LOG(DEBUG) << "Converted EMCAL cells";
+    for (auto& cell : mEmcalCells) {
+      LOG(DEBUG) << cell.getTower() << ": E: " << cell.getEnergy() << ", time: " << cell.getTimeStamp() << ", type: " << cell.getType();
+    }
+
+    LOG(INFO) << "Converted cells. Contains: " << mEmcalCells.size() << ". Originally " << cells.size() << ". About to run clusterizer.";
+
+    // Run the clusterizer
+    mClusterizer->findClusters(mEmcalCells);
+    LOG(DEBUG) << "Found clusters.";
+    auto emcalClusters = mClusterizer->getFoundClusters();
+    auto emcalClustersInputIndices = mClusterizer->getFoundClustersInputIndices();
+    LOG(DEBUG) << "Retrieved results. About to setup cluster factory.";
+
+    // Convert to analysis clusters.
+    // First, the cluster factory requires cluster and cell information in order to build the clusters.
+    mAnalysisClusters.clear();
+    mClusterFactory->reset();
+    mClusterFactory->setClustersContainer(*emcalClusters);
+    mClusterFactory->setCellsContainer(mEmcalCells);
+    mClusterFactory->setCellsIndicesContainer(*emcalClustersInputIndices);
+    LOG(DEBUG) << "Cluster factory set up.";
+
+    // Convert to analysis clusters.
+    for (int icl = 0; icl < mClusterFactory->getNumberOfClusters(); icl++) {
+      auto analysisCluster = mClusterFactory->buildCluster(icl);
+      mAnalysisClusters.emplace_back(analysisCluster);
+    }
+    LOG(DEBUG) << "Converted to analysis clusters.";
+
+    // Store the clusters in the table
+    for (const auto& cluster : mAnalysisClusters) {
+      // Determine the cluster eta, phi, correcting for the vertex position.
+      auto pos = cluster.getGlobalPosition();
+      pos = pos - math_utils::Point3D<float>{collision.posX(), collision.posY(), collision.posZ()};
+      // Normalize the vector and rescale by energy.
+      pos /= (cluster.E() / std::sqrt(pos.Mag2()));
+
+      // We have our necessary properties. Now we store outputs
+      //LOG(DEBUG) << "Cluster E: " << cluster.E();
+      clusters(collision, cluster.E(), pos.Eta(), pos.Phi(), cluster.getM02());
+      // TEMP
+      if (cluster.E() < 0.150) {
+        continue;
```

Comment:
But anyway store it in clusters() ?

Reply:
Clusters below a threshold probably aren't meaningful, so I had this in place for debugging the QA. But since this value can vary by task, it's better to leave this to the user to select, so I removed it here